packages/test-utils/src/storage-browser/mock-handlers.ts

```diff
@@ -137,17 +139,21 @@ export class MockHandlers {
       });
     }
 
-    if (
-      preventOverwrite &&
-      this.#locationItems[prefix].some((item) => item.key === key)
-    ) {
-      return {
-        ...UNDEFINED_CALLBACKS,
-        result: Promise.resolve({
-          error: new Error('cannot overwrite existing file'),
-          status: 'OVERWRITE_PREVENTED',
-        }),
-      };
+    if (this.#locationItems[prefix].some((item) => item.key === key)) {
+      if (preventOverwrite) {
+        return {
+          ...UNDEFINED_CALLBACKS,
+          result: Promise.resolve({
+            error: new Error('cannot overwrite existing file'),
+            status: 'OVERWRITE_PREVENTED',
+          }),
+        };
+      } else {
+        // remove prev item as new version will be added, 'overwriting' prev item
+        this.#locationItems[prefix] = this.#locationItems[prefix].filter(
```

Comment:
Sanity check, the overwritten file will be added to new array in different order, will it change how they are a displayed in the UI?